src/core/search/vector.cc

```diff
@@ -0,0 +1,24 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/vector.h"
+
+#include <cmath>
+
+#include "base/logging.h"
+
+namespace dfly::search {
+
+using namespace std;
+
+// Euclidean vector distance: sqrt( sum: (u[i] - v[i])^2  )
+float VectorDistance(const FtVector& u, const FtVector& v) {
```

Comment:
Otherwise GCC doesn't vectorize the loop -

```suggestion
__attribute__((optimize("fast-math")))
float VectorDistance(const FtVector& u, const FtVector& v) {
```